frontend/src/components/common/Resource/EditorDialog.tsx

```diff
@@ -152,48 +164,51 @@ export default function EditorDialog(props: EditorDialogProps) {
     );
 
     setCode({ code: value as string, format: code.format });
+    if (error && getObjectsFromCode({ code: value as string, format: originalCode.format })) {
+      setError('');
+    }
 
     if (onEditorChanged) {
       onEditorChanged(value as string);
     }
   }
 
-  function getObjectFromCode(codeInfo: typeof originalCode) {
-    function looksLikeJson(code: string) {
-      const trimmedCode = code.trimRight();
-      const lastChar = !!trimmedCode ? trimmedCode[trimmedCode.length - 1] : '';
-      const firstChar = !!trimmedCode ? trimmedCode[0] : '';
-      if (['{', '['].includes(firstChar) || ['}', ']'].includes(lastChar)) {
-        return true;
-      }
-
-      return false;
-    }
-
+  function getObjectsFromCode(codeInfo: typeof originalCode): {
+    obj: KubeObjectInterface[] | null;
+    format: string;
+    error: Error | null;
+  } {
     const { code, format } = codeInfo;
-    const res: { obj: KubeObjectInterface | null; format: string; error: Error | null } = {
+    const res: { obj: KubeObjectInterface[] | null; format: string; error: Error | null } = {
       obj: null,
       format,
       error: null,
     };
 
-    if (!res.obj) {
-      res.format = 'yaml';
+    if (!format || (!res.obj && looksLikeJson(code))) {
+      res.format = 'json';
```

Comment:
I don't understand why this switched the order of processing YAML first and then JSON (I think this way we may attempt to convert to JSON all the time when the user may try YAML first).